components/sync/engine/brave_syncer_device_poll.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_SYNC_ENGINE_BRAVE_SYNCER_DEVICE_POLL_H_
+#define BRAVE_COMPONENTS_SYNC_ENGINE_BRAVE_SYNCER_DEVICE_POLL_H_
+
+#include "base/functional/callback_forward.h"
+#include "base/gtest_prod_util.h"
+#include "base/time/time.h"
+
+namespace syncer {
+
+FORWARD_DECLARE_TEST(BraveSyncerDevicePoll, ForcedPolling);
+
+class BraveSyncerDevicePoll {
+ public:
+  BraveSyncerDevicePoll(const BraveSyncerDevicePoll&) = delete;
+  BraveSyncerDevicePoll& operator=(const BraveSyncerDevicePoll&) = delete;
+  BraveSyncerDevicePoll();
+
+  void CheckIntervalAndPoll(base::OnceClosure poll_devices);
+  base::TimeDelta SinceBegin();
+
+ private:
+  FRIEND_TEST_ALL_PREFIXES(BraveSyncerDevicePoll, ForcedPolling);
+  static base::TimeDelta GetDelayBeforeForcedPollForTest();
```

Comment:
nit: `ForTesting` is more widely used suffix.
![image](https://user-images.githubusercontent.com/5928869/217479337-5a2f1a36-af58-4244-89f7-ce25b61d8317.png)
![image](https://user-images.githubusercontent.com/5928869/217479352-70cc4bcf-9869-42d9-8bef-c146fe550e9e.png)

Reply:
thanks for the investigation and reasoning, fixed at https://github.com/brave/brave-core/pull/17077/commits/d8d6a58998131d2e616c14dc6757d70c5d0a232f